src/System.CommandLine/Binding/ModelBinder.cs

```diff
@@ -219,13 +212,12 @@ public void UpdateInstance<T>(T instance, BindingContext bindingContext)
                         {
                             if (parameterDescriptor.HasDefaultValue)
                                 boundValue = BoundValue.DefaultForValueDescriptor(parameterDescriptor);
-                            else if (parameterDescriptor.AllowsNull && 
+                            else if (parameterDescriptor.AllowsNull &&
                                 ShouldPassNullToConstructor(constructorDescriptor.Parent, constructorDescriptor))
                                 boundValue = BoundValue.DefaultForType(valueDescriptor);
```

Comment:
It looks like this block should also be part of the extracted `GetBoundValue` method.